### ISSUE TITLE
Metrics first iteration

### ISSUE THREAD

**Issue by moolen** [external-secrets/external-secrets#30]
As a Cluster Operator i need metrics/alerts to ensure ESO runs smoothly, secrets are beings synced and SecretStores are properly configured

AC:
* outline/document/implement MVP metrics for the operator ([kubebuilder docs](https://book.kubebuilder.io/reference/metrics.html))

**Comment by moolen** [comment]
User Story:

I as a cluster admin want to ensure, that ES are continuously synced and that i get an alert if a sync of an ES fails. 
I think the following things might help a cluster admin:
* sync_calls_counter that tracks the number of sync calls per ES with name, namespace as label (maybe a storeRef?)
* sync_failed_counter with labels: name, namespace (or maybe as a status label on the above metric)
* (maybe) having an count of api calls per provider (to keep track of $$$ consumption)
* example alerts would be really great
* also having a grafana dashboard blueprint would be nice
* I don’t know whats included in the kubebuilder default metrics and if we can configure the port, that would be nice to have documentation for

**Comment by jonatasbaldin** [comment]
hey peeps, I'll work start working on this, seems interesting!

**Comment by moolen** [comment]
I think @mircea-cosbuc already started to take a look at it, i think?!

**Comment by mircea-cosbuc** [comment]
Hey @jonatasbaldin , I did start looking into this and I was testing out the metrics exposed by default. We can sync if you'd like and check if we want to work on it together.

**Comment by jonatasbaldin** [comment]
yeah, let's do it :)

**Comment by jonatasbaldin** [comment]
Questions:

1. The `kubebuilder` generates manifests with side-car container/service/RBAC (`config/default/manager_auth_proxy_patch.yaml` etc) to access the `/metrics` endpoint with authentication/TLS. By removing these manifests (#92) (and not integrating them into the Helm chart) we are losing this functionality. Do we want to keep this auth functionality or will we just expose the Prometheus metrics by creating a Service on port 8080?

2. If we decide to export the `/metrics` without any auth/TLS, we can also let Prometheus scrape them in the [Pod (example from cert-manager)](https://github.com/jetstack/cert-manager/blob/master/deploy/charts/cert-manager/templates/deployment.yaml#L44) instead of creating the `deploy/charts/external-secrets/templates/service.yaml` Service. What do you think would be best?

**Comment by moolen** [comment]
I think we discussed your questions that in the community meeting, right? 
tl;dr IIRC: we don't support direct TLS or proxy-sidecars

Should we keep this issue open to track further enhancements or create a new issue for that?

**Comment by knelasevero** [comment]
If we open another issue we can link to this one. Are we planning on supporting direct TLS or proxy-sidecars anytime soon?

@mircea-cosbuc  was also working on something here. @mircea-cosbuc can you list the things that are being done by you here? Or were they covered already?

**Comment by mircea-cosbuc** [comment]
@knelasevero I want to add a counter of the provider specific api calls for helping with cost management.

**Comment by moolen** [comment]
I created a follow-up issue #648 for this use-case
